packages/runtime/container-runtime/src/compatUtils.ts

```diff
@@ -0,0 +1,204 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils/internal";
+import { FlushMode } from "@fluidframework/runtime-definitions/internal";
+import * as semver from "semver";
+
+import {
+	type ICompressionRuntimeOptions,
+	type IContainerRuntimeOptionsInternal,
+} from "./containerRuntime.js";
+import { pkgVersion } from "./packageVersion.js";
+
+/**
+ * List of all 2.x versions where the FF runtime's disallowedVersions implementation uses exact versions.
+ * TODO: How do we maintain this list when patches are published? We don't want to fetch these versions from the registry since that will be slow.
+ */
+const packages2x = [
+	"2.0.0",
+	"2.0.1",
+	"2.0.2",
+	"2.0.3",
+	"2.0.4",
+	"2.0.5",
+	"2.1.0",
+	"2.0.6",
+	"2.2.0",
+	"2.3.0",
+	"2.0.7",
+	"2.1.1",
+	"2.2.1",
+	"2.0.8",
+	"2.0.9",
+	"2.1.2",
+	"2.2.2",
+	"2.3.1",
+	"2.4.0",
+	"2.5.0",
+	"2.10.0",
+	"2.11.0",
+	"2.12.0",
+	"2.13.0",
+	"2.20.0",
+	"2.21.0",
+	"2.22.0",
+	"2.22.1",
+	"2.23.0",
+	"2.30.0",
+];
+
+// TODO: CompressionAlgorithms and disabledCompressionConfig are copied here to avoid weird circular dependency issue
+
+/**
+ * Available compression algorithms for op compression.
+ * @legacy
+ * @alpha
+ */
+export enum CompressionAlgorithms {
+	lz4 = "lz4",
+}
+
+/**
+ * @legacy
+ * @alpha
+ */
+export const disabledCompressionConfig: ICompressionRuntimeOptions = {
+	minimumBatchSizeInBytes: Number.POSITIVE_INFINITY,
+	compressionAlgorithm: CompressionAlgorithms.lz4,
+};
+
+export const enabledCompressionConfig = {
+	// Batches with content size exceeding this value will be compressed
+	minimumBatchSizeInBytes: 614400,
+	compressionAlgorithm: CompressionAlgorithms.lz4,
+};
+
+const defaultFlushMode = FlushMode.TurnBased;
+
+/**
+ * Subset of the IContainerRuntimeOptionsInternal properties which are version-dependent.
+ */
+type IContainerRuntimeOptionsVersionDependent = Pick<
```

Comment:
Does this mean developers have to add a container runtime option in two places - Just the name: value in `IContainerRuntimeOptionsInternal` and then add it here with the compat options? Wouldn't that increase the chances of these two diverging?
Have you considered requiring adding this to just one place or have some kind of validatiion to ensure these two are in sync?

Reply:
Also, how do you think it can be ensured that a min version is added when a new container runtime option is added? Should that be done via automated mechanism or validation or just guidelines?

Reply:
This is a good point. We will certinaely have guidelines for developers on how to properly introduce new options, but I will also look into adding some type of mechanism to make this clear when adding new properties to `IContainerRuntimeOptions`.